textcursor.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gxui
+
+import "github.com/google/gxui/interval"
+
+// TextCursor represents a text position and selection. A selection is defined as
+// the pair of cursor indices sorted least to greatest. When transforming a
+// cursor with no selection, mark should equal index.
+type TextCursor struct {
```

Comment:
This makes the construction of thr non-selection case simple:
```go
gxui.TextCursor{ Index: i }
```

`t = t.Offset(i)` becomes `t.Index += i`

Things that don't want to know about selections don't look at the `Length` field. 

My concerns with half-closed intervals go away as it's much simpler to understand.